src/Step/Deploy/InstallUpdate/ConfigUpdate/CacheType.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Step\Deploy\InstallUpdate\ConfigUpdate;
+
+use Magento\MagentoCloud\App\GenericException;
+use Magento\MagentoCloud\Config\Stage\DeployInterface;
+use Magento\MagentoCloud\Filesystem\Flag\Manager as FlagManager;
+use Magento\MagentoCloud\Shell\MagentoShell;
+use Magento\MagentoCloud\Shell\ShellException;
+use Magento\MagentoCloud\Shell\ShellFactory;
+use Magento\MagentoCloud\Step\StepException;
+use Magento\MagentoCloud\Step\StepInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class CacheType implements StepInterface
+{
+    /**
+     * @var FlagManager
+     */
+    private $flagManager;
+
+    /**
+     * @var DeployInterface
+     */
+    private $stageConfig;
+
+    /**
+     * @var MagentoShell
+     */
+    private $magentoShell;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param FlagManager $flagManager
+     * @param DeployInterface $stageConfig
+     * @param ShellFactory $shellFactory
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        FlagManager $flagManager,
+        DeployInterface $stageConfig,
+        ShellFactory $shellFactory,
+        LoggerInterface $logger
+    ) {
+        $this->flagManager = $flagManager;
+        $this->stageConfig = $stageConfig;
+        $this->magentoShell = $shellFactory->createMagento();
+        $this->logger = $logger;
+    }
+
+    /**
+     * Enable all cache types if env.php was absence
+     *
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        try {
+            if ($this->flagManager->exists(FlagManager::FLAG_ENV_FILE_ABSENCE)) {
+                $this->logger->info('Run cache:enable to restore all cache types');
+                $this->magentoShell->execute(
+                    'cache:enable',
+                    [$this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS)]
+                );
+            }
+        } catch (ShellException $exception) {
+            //code will be changed after merging PR with error reporting
```

Comment:
If I understand your comment correctly, this exception will change slightly once specific error codes are created? Otherwise it seems like we should be able to just catch `GenericException`s below.

Reply:
yes, as the codes for theres exceptions will be different I caught both types, just not to forget about it after merging PR https://github.com/magento/ece-tools/pull/726/, e.g. https://github.com/magento/ece-tools/pull/726/files#diff-11ebfae7dd9722de4d90062c73b5170fR89